### PR TITLE
React & Redux group project - Space Travelers' Hub

### PR DESCRIPTION
**Pull request by mariordgez**
In this Project, we worked with real live data from the SpaceX API. We build a web application for a company that provides commercial and scientific space travel services. The application will allow users to book rockets, dragons and join selected space missions.

Rockets /Dragons

The Rockets section displays a list of all available SpaceX rockets. Users can book each rocket by clicking the reservation button or cancel the previously made booking. The same layout is used to form the Dragons section Space Travelers Hub.
Here is the wire-frame for the Rockets/Dragons page:
image

Missions

The Missions section displays a list of current missions along with their brief description and participation status. There is also a button next to each mission that allows users to join the selected mission or leave the mission the user joined earlier. 

image

My Profile

The My Profile section displays all reserved rockets, dragons and space missions.
We made sure to only dispatch the fetch actions once and do not add data to store on every re-render (i.e. when changing views / using navigation).

Hope you like our project and have a wonderful day!!

### CODE REVIEW

**Review by mke2111 — CHANGES_REQUESTED** [review]
### Hey @mariordgez & @Emmanuelaaron & @enionsouza, 

**This is a great project you have going on, well-done lads**.  Most of your implementations are spot-on. Cheers! 👍🏽 💯 

   
**However, there are a few sections to improve before your project is approved.** 🔴 
<hr>

## Required Changes ♻️
_Check the inline comments under the review for sections to improve._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Missions.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+                  <Badge bg="secondary">NOT A MEMBER</Badge>
+                )}
+                {' '}
+              </td>
+              <td className="align-middle">
+                <JoinLeaveBtn reserved={mission.reserved} id={mission.mission_id} />
+              </td>
+            </tr>
+          ))}
+        </tbody>
```
Comment (mke2111) [inline src/components/Missions.js]: - [x] For this group project, Rockets/Dragons and Missions should use the React conditional rendering syntax. You have implemented this perfectly in the components/Dragons.js and in components/Rockets.js, but it is missing in the components/Missions.js file. You could implement this in many ways, one of them being:
`{missions && missions.map((mission) => (`   --------------on line 28

File: src/components/MyProfile.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+          .filter((dragon) => dragon.reserved === true)
+          .map((filteredDragon) => (
+            <ListGroup key={filteredDragon.id}>
+              <ListGroup.Item>{filteredDragon.name}</ListGroup.Item>
+            </ListGroup>
+          ))}
+      </Card>
+    </Container>
+  );
+};
```
Comment (mke2111) [inline src/components/MyProfile.js]: OPTIONAL

- The My Profile section can be enhanced by adding a placeholder message when the "My Missions" or "My Rockets" lists are empty (no missions joined or no rockets reserved).
- The My Profile section can be enhanced by adding the "Cancel reservation" and "Leave Mission" buttons to the lists here. Clicking them should dispatch the actions you have already used in the main Rockets and Missions sections.
- The My Profile section can be enhanced by adding the "Read more" button for each mission and rocket. Upon click, it should open a corresponding Wikipedia page in a new tab. NOTE - you need to get that extra Wikipedia URL from the API's payload.

File: README.md [hunk 3]
```diff
@@ -1,70 +1,114 @@
...
+  <a href="https://www.microverse.org/">
+    <img alt="Microverse" src="https://img.shields.io/badge/-Microverse-blueviolet?style=flat-square">
+  </a>
+  <a href="https://github.com/Emmanuelaaron/spaceships">
+    <img alt="GitHub last commit (branch)" src="https://img.shields.io/github/last-commit/Emmanuelaaron/spaceships/dev?color=blue&style=flat-square">
+  </a>
+  <a href="https://github.com/Emmanuelaaron/spaceships">
+    <img alt="GitHub Repo stars" src="https://img.shields.io/github/stars/Emmanuelaaron/spaceships?color=cyan&label=%E2%98%85%20stars%20&style=flat-square">
+  </a>
+</p>
```
Comment (mke2111) [inline README.md:11]: OPTIONAL

- [ ] Add the live link to the deployed version of your app to this file.

File: src/components/Missions.js (hunk 1 above)
Comment (mariordgez) [reply src/components/Missions.js]: Missions are now rendered using react conditional render syntax

**Review by BrianSammit — APPROVED** [review]
# STATUS: APROVED :green_circle: :clap: 

Hello team 👋, 
Your project is complete!  There is nothing else to say other than... Great job 👏 
Congratulations! 🎉 

Now you can merge this branch and move on :1st_place_medal: 
HAPPY CODING AND GOOD LUCK WITH THE COMMING ONES :crossed_fingers: :four_leaf_clover: 

200w


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**